twins/mongodb/twins.go

```diff
@@ -0,0 +1,229 @@
+//
+// Copyright (c) 2019
+// Mainflux
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package mongodb
+
+import (
+	"context"
+
+	"github.com/mainflux/mainflux/twins"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	maxNameSize            = 1024
+	twinsCollection string = "twins"
+)
+
+type twinRepository struct {
+	db *mongo.Database
+}
+
+var _ twins.TwinRepository = (*twinRepository)(nil)
+
+// NewTwinRepository instantiates a MongoDB implementation of twin
+// repository.
+func NewTwinRepository(db *mongo.Database) twins.TwinRepository {
+	return &twinRepository{
+		db: db,
+	}
+}
+
+// Save persists the twin
+func (tr *twinRepository) Save(ctx context.Context, tw twins.Twin) (string, error) {
+	if len(tw.Name) > maxNameSize {
+		return "", twins.ErrMalformedEntity
+	}
+
+	coll := tr.db.Collection(twinsCollection)
+
+	if _, err := tr.RetrieveByID(ctx, tw.ID); err == nil {
```

Comment:
when is this ID created?
doesnt the mongo driver take care that there is already entity with ID?